app/controllers/support_interface/api_tokens_controller.rb

```diff
@@ -5,8 +5,18 @@ def index
     end
 
     def create
+      raise_error_unless_provider(params)
+
       provider = Provider.find(params[:vendor_api_token][:provider_id])
       @unhashed_token = VendorAPIToken.create_with_random_token!(provider: provider)
     end
+
+  private
+
+    def raise_error_unless_provider(params)
+      if params[:vendor_api_token][:provider_id].blank?
+        raise MissingProviderError, 'Did not select a provider'
```

Comment:
I can see the error was raised in Sentry but the user will hit the 'Service unavailable' page. What are your thoughts on adding a redirect back to the provide api page if this error is raised and perhaps rendering a flash warning box with that message?

Reply:
Thanks James, I've pushed up the changes